backend/remote-state/http/backend.go

```diff
@@ -0,0 +1,301 @@
+			},
+			"local_ca_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "CA to be used for TLS",
+			},
+			"mutual_tls_authentication": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Use mutual tls authentication.local_cert_file, local_key_file, local_ca_file needs to be set. Defaults to false.",
+			},
+
+			"local_cert_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Cert file to be used for mutual tls authentication.",
+			},
+
+			"local_key_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Key file to be used for mutual tls authentication.",
+			},
+		},
+	}
+
+	result := &Backend{Backend: s}
+	result.Backend.ConfigureFunc = result.configure
+	return result
+}
+
+// Backend Structure
+type Backend struct {
+	*schema.Backend
+	data   *schema.ResourceData
+	client *http.Client
+
+	// The fields below are set from configure
+	address         string
+	updateMethod    string
+	lockAddress     string
+	lockMethod      string
+	unlockMethod    string
+	unlockAddress   string
+	skipTLS         bool
+	localCertCAFile string
+	localCertFile   string
+	localKeyFile    string
+	localCAFile     string
+	mutualTLS       bool
+	username        string
+	password        string
+}
+
+func (b *Backend) configure(ctx context.Context) error {
+	if b.client != nil {
+		return nil
+	}
+	// Grab the resource data
+	data := schema.FromContextBackendConfig(ctx)
+
+	// Store the URL address
+	addressURL := data.Get("address").(string)
+	// Validate URL address
+	validationErr := assertValidURL(addressURL)
+	if validationErr != nil {
+		return validationErr
+	}
+	b.address = addressURL
+
+	b.updateMethod = data.Get("update_method").(string)
+
+	if v, ok := data.GetOk("lock_address"); ok {
+		lockAddress := v.(string)
+		// Validate lockAddress
+		validationErr := assertValidURL(lockAddress)
+		if validationErr != nil {
+			return validationErr
+		}
+		b.lockAddress = lockAddress
+	} else {
+		// If lockAddress is null, use the http rest api address
+		b.lockAddress = b.address
+	}
+
+	b.lockMethod = data.Get("lock_method").(string)
+
+	if v, ok := data.GetOk("unlock_address"); ok {
+		unlockAddress := v.(string)
+		// Validate unlockAddress
+		validationErr := assertValidURL(unlockAddress)
+		if validationErr != nil {
+			return validationErr
+		}
+		b.unlockAddress = unlockAddress
+	} else {
+		// If unlockAddress is null, use the http rest api address
+		b.unlockAddress = b.address
+	}
+
+	b.unlockMethod = data.Get("unlock_method").(string)
+
+	if v, ok := data.GetOk("username"); ok {
+		b.username = v.(string)
+	}
+
+	if v, ok := data.GetOk("password"); ok {
+		b.password = v.(string)
+	}
+
+	client := &http.Client{
+		Timeout: time.Second * 10,
+	}
+	// skip_cert_verification client
+	if v, ok := data.GetOk("skip_cert_verification"); ok {
```

Comment:
`skip_cert_verification` is optional with a default, you can use `d.Get()` instead of `d.GetOk()` safely here which will save you some indentation:

```
if b.skipTLS = data.Get("skip_cert_verification").(bool); b.skipTLS {
  // ...
}
```